dask/array/core.py

```diff
@@ -1906,6 +1906,36 @@ def __setitem__(self, key, value):
         self._name = y.name
         self._chunks = y.chunks
 
+    def bool_to_int_index(self, a):
+        """Transforms a bool array index to int array index for 1D arrays where
+        all the False values are adjacent and are either at the beginning or end
+        of the array.
+        """
+        from dask.array.creation import arange
+
+        if self.ndim != 1 or math.isnan(a[0].size) or math.isnan(self.shape[0]):
+            return None
+
+        x = 0, a[0].argmax()
+        # Checks if all False values are in the beginning of the array.
+        for i in range(a[0].size - 1):
+            if a[0][i] > a[0][i + 1]:
```

Comment:
Thanks @geraninam! FWIW I have some reservations about this implementation as it appears to rely on some current behavior where we will trigger an implicit computation

https://github.com/dask/dask/blob/1fcfb7c93e83908eee4449b473f6e325b79e1f1a/dask/array/core.py#L1832-L1839

From what I can tell, it looks like `bool_to_int_index` can trigger many implicit computations due to these types of comparisons happening in a `for`-loop. 

I'm not sure that we should have allowed `Array.__bool__` to have an implicit `compute` in the first place...I'm seeing how much other behavior in Dask is relying on this over in https://github.com/dask/dask/pull/9387

Reply:
Ah that is a very good point. I should have caught that in earlier reviews. Sorry about that @geraninam and @dustinwerran2

Reply:
We're watching the discussions in the other thread about this. If we can resolve something today or tomorrow, we might have time to get the fix in before the mentorship program ends on Friday :)